src/game/etj_timerun_v2.cpp

```diff
@@ -1006,58 +1012,73 @@ void ETJump::TimerunV2::checkRecord(Player *player) {
         for (const auto &tr : topRecords) {
-          seasonIdToTopRecord[tr.seasonId] = tr;
+          topRecordForSeason[tr.seasonId] = tr;
         }
+
         auto playerRecords = _repository->getRecordsForPlayer(
             _activeSeasonsIds, _currentMap, activeRunName, userId);
 
-        std::map<int, Timerun::Record> seasonIdToNewRecord{};
-        std::map<int, const Timerun::Record *> seasonIdToPreviousRecord{};
+        std::map<int, const Timerun::Record *> playerTopRecordForSeason{};
+
+        // get our previous top times of the run we just finished
+        // for active seasons
         for (const auto &r : playerRecords) {
-          seasonIdToPreviousRecord[r.seasonId] = &r;
+          playerTopRecordForSeason[r.seasonId] = &r;
         }
 
-        std::map<int, bool> seasonIdToIsNewRecord{};
+        std::map<int, Timerun::Record> playerNewTopRecordForSeason{};
+        std::map<int, bool> isNewRecordForSeason{};
+
+        // go through active seasons and check if we beat any of the old records
+        // if we did, we store the record in playerNewTopRecordForSeason
+        // this only checks against our own personal previous times,
+        // not against other player's times
         for (auto seasonId : _activeSeasonsIds) {
           Timerun::Record record;
           record.seasonId = seasonId;
           record.map = _currentMap;
           record.run = activeRunName;
           record.userId = userId;
-          record.time = completionTime.value();
+          record.time = completionTime;
           record.checkpoints = checkpoints;
           record.recordDate = getCurrentTime();
           record.playerName = playerName;
           record.metadata = metadata;
 
           auto insert = false;
           auto isNewRecord = false;
-          if (seasonIdToPreviousRecord.count(seasonId) == 0) {
+          if (playerTopRecordForSeason.count(seasonId) == 0) {
             isNewRecord = true;
             insert = true;
           } else {
             insert = false;
-            isNewRecord = completionTime.value() <
-                          seasonIdToPreviousRecord[seasonId]->time;
+            isNewRecord =
+                completionTime < playerTopRecordForSeason[seasonId]->time;
           }
-          seasonIdToIsNewRecord[seasonId] = isNewRecord;
+          isNewRecordForSeason[seasonId] = isNewRecord;
 
           if (isNewRecord) {
             if (insert) {
               _repository->insertRecord(record);
             } else {
               _repository->updateRecord(record);
             }
-            seasonIdToNewRecord[seasonId] = std::move(record);
+            playerNewTopRecordForSeason[seasonId] = std::move(record);
           }
         }
 
         auto result = std::make_unique<CheckRecordResult>();
         result->clientNum = clientNum;
 
-        for (const auto &s : seasonIdToIsNewRecord) {
+        // setup new records for any of our old records we beat
```

Comment:
I actually have no idea what this comment means :D

Reply:
Yeah now looking at it it's not great I guess, better comment would be I guess "Go through our old records and update any record that we beat"